src/tfp_backend/Code_gen.ml

```diff
@@ -193,8 +193,11 @@ let get_params p =
 
 let pp_shapes ppf p =
   let pp_shape ppf (_, {Program.out_unconstrained_st; _}) =
-    pf ppf "(nchains__, @[<hov>%a@])" (list ~sep:comma pp_expr)
-      (SizedType.get_dims out_unconstrained_st)
+    let shapes_lst = SizedType.get_dims out_unconstrained_st in
+    if List.length shapes_lst > 0 then
+      pf ppf "(nchains__, tf__.cast(@[<hov>%a@], tf__.int32))"
+        (list ~sep:comma pp_expr) shapes_lst
+    else pf ppf "(nchains__, )"
```

Comment:
Is it possible for there to be more than one thing in `shapes_lst`? If so that would give `tf__.cast(J, K, L, tf__.int32)` which seems to not work for tf.cast.